### DISCUSSION TITLE
JVM Client Library

### DISCUSSION THREAD

**Discussion by davidjayb** [zitadel/zitadel#3650] [Contributors]
Hey folks,

I am interested in trying my hand at building a JVM client library for Zitadel. Are there any docs for building a client library, such as the APIs required, or would it be best to reference existing libraries?

Thanks,
David

**Comment by fforootd** [comment]
Thank for you asking. 

@fgerschwiler I think you were coordinating something on this end, right?

**Comment by hifabienne** [comment]
Hi @davidjayb 

Thank you very much. 
I will have a chat this Friday, with someone who wants to help us with Spring Boot integration.
Not sure yet what het will do.
I will reach out to you after that if thats ok.

At the moment we don't have docs about how to build a client library. So the best would be to look at the other libs we already have.
In my opinion there are the following things that are important:
- Authentication
- Authorization
- Calling ZITADEL APIs

**Comment by hifabienne** [comment]
Hi @davidjayb 

It would be great if you could have a look at the JVM SDK.
We have two sdks at the moment. One for golang and one for .NET
I think the easiest way is to have a look at them:
https://github.com/smartive/zitadel-net

It would be great if the SDK has all the configurations needed to authenticate a user and check the roles of them. And also to make some requests against our API. (With a service user and jwt profile)
The SDKs should generate a client based on the proto files we have in the ZITADEL repo.
https://github.com/zitadel/zitadel/tree/v2-alpha/proto/zitadel

I think the docs of the .NET SDK can help you: 
https://github.com/smartive/zitadel-net/tree/main/src/Zitadel
https://github.com/smartive/zitadel-net/tree/main/src/Zitadel.Grpc

If you need some more informations just let me know. Or we can also make a quick call to discuss this.

Thanks in advanced.

**Reply by esafak** [reply]
The doc links are broken. Did they show how to use the gRPC API?

**Comment by MuizMahdi** [comment]
Any update on this @davidjayb?
